runtime/util/annhelp.c

```diff
@@ -99,24 +99,40 @@ skipAnnotationElement(J9ROMConstantPoolItem const * constantPool, U_8 const *dat
 	case 'S':
 	case 'Z':
 	case 's':
-	case 'c':
+	case 'c': {
+		U_16 constValueOrClassInfoIndex = 0;
```

Comment:
since you are not actually using the read value `constValueOrClassInfoIndex` you could just `index += 2`.

Or better, rename `U_8 const *index` to `data` and you wont need most of these changes

Reply:
CHECK_EOF requires both vars